### DISCUSSION TITLE
Account connecting(or linking) feature implement

### DISCUSSION THREAD

**Discussion by ChiuMungZitAlexander** [nextauthjs/next-auth#7877] [Help]
I am creating an dApp by the help of `next-auth`.

### We only allow users to sign in with Ethereum/Polygon/Sui. After the user has session, there should be a user record, as well as an account record in database as below.

User collection
```
{
  _id: ObjectId('649958e99559a97d56d2e638'),
  name: "0x3e1edd41467a458cda51110333e34a0f9307d4fe874cee4bf3b78044809cc55a"
}
```

Account collection
```
{
  _id: ObjectId('649958e99559a97d56d2e69'),
  userId: ObjectId('649958e99559a97d56d2e68'),
  type: "credentials",
  provider: "Ethereum"
}
```

### We also have a setting page to let the user connect Twitter and Discord. Then the user will be able to login by the connected Twitter or Discord account. The page where the user connects Twitter or Discord must be with a session.

I just found that if I directly call `signIn` from `next-auth/react` to connect the user's Twitter or Discord account, it will just create a new account record, but not a new user record.

User collection, **NO CHANGE**
```
{
  _id: ObjectId('649958e99559a97d56d2e638'),
  name: "0x3e1edd41467a458cda51110333e34a0f9307d4fe874cee4bf3b78044809cc55a"
}
```

Account collection, **A NEW ACCOUNT**
```
{
  _id: ObjectId('649958e99559a97d56d2e69'),
  userId: ObjectId('649958e99559a97d56d2e68'),
  type: "credentials",
  provider: "Ethereum"
},
{
  _id: ObjectId('64995a309559a97d56d2e63a'),
  userId: ObjectId('649958e99559a97d56d2e68),
  type: "oauth",
  provider: "discord"
}
```

It's just what I want, but I wonder if it is the recommended or common practice.

Cheers!

**Comment by yenicelik** [comment]
facing the same problem right now

**Comment by yenicelik** [comment]
@ChiuMungZitAlexander were you able to resolve this?
